[Refactoring] DAO 클래스 리팩토링

leeyounho opened the pull request:
가독성을 높이기 위해 반복되는 코드 함수화하였습니다
+ 실행 Jar파일 업로드도 하였습니다!

leeyounho committed:
[Refactoring] DAO함수 메소드 추출하여 가독성↑
공통으로 사용할 수 있는 Utility 클래스 생성.
DAO 내부에서 index 검색하는 함수를 public -> private으로 변경

leeyounho committed:
[Refactoring] add함수를 addIndex함수별로 실행되도록 리팩토링하여 코드 가독성을 높이려 했습니다.

leeyounho committed:
[Refactoring] delete, modify함수를 Index함수별로 실행되도록 리팩토링하여 코드 가독성을 높이려 했습니다.

leeyounho committed:
[Refactoring] enum 함수에 대한 nullSafeToString 함수를 생성하였습니다.

leeyounho committed:
Merge branch 'main' of https://github.com/Jaeyeong-Lee/Hawkeyes into develop_yh

 Conflicts:
	src/com/samsung/database/EmployeeDAO.java

leeyounho committed:
[Mod] 실행 Jar 파일 업데이트

leeyounho commented:
DAO 클래스의 search 함수에 대해서 원경님이 적용한 formatter가 제껄로 업데이트 되어버렸는데..
제가 search함수를 다 지우고 한개로 만들거라서 commit해두었어요
다음 리팩토링 때, search 함수를 통일하겠습니다!

Jaeyeong-Lee reviewed (approved):
Null Check를 Enum으로 둔 부분이 책임 분배 잘 된것 같고 메소드 추출로 가독성 향상이 되었다고 생각합니다 💯

leeyounho committed:
[Refactoring] DAO search함수 리팩토링 및 실행 Jar 파일 업데이트

leeyounho commented:
> 

search함수도 리팩토링하여 올렸습니다😊

cointreau commented on src/com/samsung/database/EmployeeDAO.java:
```diff
@@ -13,51 +16,30 @@
-            employeeTable.getCertiIndex()
-                    .computeIfAbsent(employee.getCerti().toString(), useless -> new HashSet<>())
-                    .add(employee);
+            addIndex(employeeTable.getEmployeeNumberIndex(), employee.getEmployeeNumber(), employee);
```
깔끔한 함수화네요 :)

cointreau commented on src/com/samsung/database/EmployeeDAO.java:
```diff
@@ -290,278 +196,47 @@ public int add(Employee employee) {
+                modifyIndex(employeeTable.getMonthOfBirthIndex(), asIs.getMonthOfBirth(), toBe.getMonthOfBirth(), asIs, toBe);
+                modifyIndex(employeeTable.getDayOfBirthIndex(), asIs.getDayOfBirth(), toBe.getDayOfBirth(), asIs, toBe);
+                modifyIndex(employeeTable.getCareerLevelIndex(), asIs.getCareerLevel().toString(), toBe.getCareerLevel().toString(), asIs, toBe);
+                modifyIndex(employeeTable.getCertiIndex(), asIs.getCerti().toString(), toBe.getCerti().toString(), asIs, toBe);
```
같은 파일 내 다른 곳에선, 계속 Enum 값에 대해 Certi.nullSaferToString(employeeToDelete.getCerti()) 를 사용하셨는데 여기선 그냥 toString() 을 사용하신 이유가 궁급합니다~

leeyounho replied:
이부분은 제가 놓쳤네요. 마찬가지로 nullSafe함수를 사용하도록 고치겠습니다~

leeyounho committed:
[Refactoring]
CommandFactory, Utility 클래스 삼항연산자 사용 부분 간략화
Employee 클래스 미사용함수 제거
SearchOption 클래스 미사용함수 제거 및 주석 제거
EmployeeTable 클래스가 가지고 있는 table객체(싱글톤)를 final로 지정
main 함수의 의미없는 throw 구문 제거

cointreau reviewed (approved):
엄청난 양의 리팩토링이라 읽는데만도 한참걸렸네요. 고생 정말 많으셨습니다!

leeyounho committed:
[Refactoring] Enum함수 toString을 nullSafe하게 사용하도록 수정

leeyounho commented:
피드백주신 부분 반영해서 다시 commit했습니다👍
